File: implementations/go/examples/internal/application/command/withdraw_handler.go
package command

import (
	"context"
	"fmt"

	"github.com/example/account-service/internal/domain/account"
)

type WithdrawCommand struct {
	AccountID   string
	RequesterID string
	Amount      int64
}

type WithdrawHandler struct {
	repo account.Repository
}

func NewWithdrawHandler(repo account.Repository) *WithdrawHandler {
	return &WithdrawHandler{repo: repo}
}

// Handle saves and returns immediately — publishing to/consuming from SQS via
// the Outbox is solely the responsibility of the independently, periodically
// running outbox.Poller/outbox.Consumer (no synchronous draining,
// domain-events.md).
func (h *WithdrawHandler) Handle(ctx context.Context, cmd WithdrawCommand) (*account.Transaction, error) {
	a, err := account.FindOne(ctx, h.repo, cmd.AccountID, cmd.RequesterID)
	if err != nil {
		return nil, fmt.Errorf("withdraw: %w", err)
	}
	// This is a withdrawal requested directly by the user, so there is no
	// referenceID (""). Only the Payment BC reaction
	// (withdraw_by_payment_handler.go) carries a correlation key.
	tx, err := a.Withdraw(cmd.Amount, "")
	if err != nil {
		return nil, err
	}
	if err := h.repo.SaveAccount(ctx, a); err != nil {
		return nil, err
	}
	return &tx, nil
}

File: implementations/fastapi/examples/src/account/application/command/withdraw_handler.py
from dataclasses import dataclass

from ...domain.errors import AccountNotFoundError
from ...domain.repository import AccountRepository
from ...domain.transaction import Transaction


@dataclass
class WithdrawCommand:
    account_id: str
    requester_id: str
    amount: int


class WithdrawHandler:
    def __init__(self, repo: AccountRepository) -> None:
        self._repo = repo

    async def execute(self, cmd: WithdrawCommand) -> Transaction:
        accounts, _ = await self._repo.find_accounts(
            page=0, take=1, account_id=cmd.account_id, owner_id=cmd.requester_id
        )
        account = accounts[0] if accounts else None
        if account is None:
            raise AccountNotFoundError(cmd.account_id)
        transaction = account.withdraw(cmd.amount)
        await self._repo.save(account)
        return transaction
